### PR TITLE
Split export extensions into 2

### LINKED ISSUE #6075: Split the export-extensions plugin into two
Reference: https://github.com/babel/proposals/issues/19

The proposal has been split for a long time now but we haven't changed it. Now's an opportunity!
There is https://github.com/babel/proposals/issues/16 and https://github.com/babel/proposals/issues/17.

- [ ] export-ns-from is moving to Stage 2 (and potentially later skipping the process but irrelevant atm)
- [ ] export-default-from is staying at Stage 1 and also having some other changes
- [ ] moving the stages is similar to https://github.com/babel/babel/pull/5877
- [ ] splitting the package in 2 however means you need to make a new folder for each, etc.

---

- [ ] Comment below you are going to do this (for others to know)
- [ ] Read [CONTRIBUTING.md](https://github.com/babel/babel/blob/7.0/CONTRIBUTING.md)
  - [ ] at minimum you'll need to run `make bootstrap` after cloning.
- [ ] Check you are on 7.0 branch with `git checkout 7.0` and then make a branch off of that to PR
- [ ] [Join our slack](http://slack.babeljs.io/) if you haven't already if you need more immediate help or just comment in this issue/PR
- [ ] make the changes above

### PR DESCRIPTION
**Pull request by echo304**
| Q                        | A 
| ------------------------ | ---
| Patch: Bug Fix?          |  no
| Major: Breaking Change?  |  yes
| Minor: New Feature?      |  no
| Deprecations?            |  no
| Spec Compliancy?         |  no
| Tests Added/Pass?        |  no
| Fixed Tickets            | Fixes #6075
| License                  | MIT
| Doc PR                   | 
| Dependency Changes       | 

https://github.com/babel/babel/issues/6075
Split `babel-plugin-transform-export-extensions` into two while keeping `babel-plugin-syntax-export-extensions`
 - `babel-plugin-transform-export-default`
 - `babel-plugin-transform-export-namespace`

### CODE REVIEW

**Comment by hzoo** [comment]
> while keeping babel-plugin-syntax-export-extensions

Actually thats a good point, we may want to figure that out too (in a later pr/change)

**Comment by hzoo** [comment]
Sorry will need to do a rebase/fix conflicts since we modified class properties in https://github.com/babel/babel/pull/6076

**Comment by echo304** [comment]
rebased and fixed conflicts!

File: packages/babel-plugin-transform-export-default/README.md [hunk 1]
```diff
@@ -1,18 +1,17 @@
-# babel-plugin-transform-export-extensions
+# babel-plugin-transform-export-default
 
-> Compile additional export-from statements to ES2015
+> Compile additional export-default-from statements to ES2015
```
Comment (existentialism) [inline packages/babel-plugin-transform-export-default/README.md]: Nit: can probably drop the "additional" bit?

File: packages/babel-preset-stage-2/package.json [hunk 2]
```diff
@@ -11,6 +11,7 @@
     "babel-plugin-transform-class-properties": "7.0.0-alpha.19",
     "babel-plugin-transform-function-sent": "7.0.0-alpha.19",
     "babel-plugin-transform-numeric-separator": "7.0.0-alpha.19",
+    "babel-plugin-transform-export-namespace": "7.0.0-alpha.19",
```
Comment (existentialism) [inline packages/babel-preset-stage-2/package.json]: Nit: alpha sort this

**Comment by hzoo** [comment]
If you want we could make the old name use the 2 new plugins and deprecate it to say you should use the new one but dono if we want to do that

**Comment by echo304** [comment]
@hzoo Let me clarify a little bit. So what you've suggested is that we'd better remain old plugin named `babal-plugin-transform-export-extensions` which refer to 2 new plugins(`export-namespace` and `export-default`) to support backward compatibility. Am I right?

**Comment by hzoo** [comment]
Thanks @echo304, this is great!

Yeah I was thinking we the old plugin would just include both but we would deprecate it (we don't have to do this). Or we just make the new plugin error and say you have to include either of the new plugins.

**Pull request merged**